### PR TITLE
Fix a bunch of 64-bit MSVC warnings

### PR DESCRIPTION
**Pull request by asarium**

### CODE REVIEW

File: code/ai/aibig.cpp [hunk 1]
```diff
@@ -917,7 +917,7 @@ void ai_big_chase()
 	} else if (En_objp->flags & OF_PROTECTED) {	//	If protected and we're not attacking a subsystem, stop attacking!
 		update_aspect_lock_information(aip, &vec_to_enemy, dist_to_enemy - En_objp->radius, En_objp->radius);
 		aip->target_objnum = -1;
-		if (find_enemy(Pl_objp-Objects, MAX_ENEMY_DISTANCE, The_mission.ai_profile->max_attackers[Game_skill_level]) == -1) {
+		if (find_enemy(OBJ_INDEX(Objects), MAX_ENEMY_DISTANCE, The_mission.ai_profile->max_attackers[Game_skill_level]) == -1) {
```
Comment (MageKing17) [inline code/ai/aibig.cpp]: I'm pretty sure this should be `OBJ_INDEX(Pl_objp)` rather than `OBJ_INDEX(Objects)`.

Comment (asarium) [inline code/ai/aibig.cpp]: Good catch, I'll fix this right away.

File: code/cfile/cfilearchive.cpp [hunk 2]
```diff
@@ -116,12 +115,13 @@ int cftell( CFILE * cfile )
...
-		int raw_position;
-		raw_position = ftell(cb->fp) - cb->lib_offset;
-		Assert(raw_position == cb->raw_position);
+	auto raw_position = ftell(cb->fp) - cb->lib_offset;
+	Assert(raw_position == cb->raw_position);
 	#endif
 
-	return cb->raw_position;
+	// The rest of the code still uses ints, do an overflow check to detect cases where this fails
+	Assertion(cb->raw_position <= std::numeric_limits<size_t>::max(), "Integer overflow in cftell, a file is probably too large (but I don't know which one).");
```
Comment (MageKing17) [inline code/cfile/cfilearchive.cpp]: I'm not sure I follow the logic of this Assertion. How does comparing with the maximum `size_t` guard against an `int` overflow?

Comment (asarium) [inline code/cfile/cfilearchive.cpp]: Yeah, that should be an `int`.

File: code/cfile/cfile.cpp [hunk 3]
```diff
@@ -1423,7 +1421,10 @@ int cfilelength( CFILE * cfile )
 	Assert(cb->fp != NULL);
 
 	// cb->size gets set at cfopen
-	return cb->size;
+	
+	// The rest of the code still uses ints, do an overflow check to detect cases where this fails
+	Assertion(cb->size <= std::numeric_limits<size_t>::max(), "Integer overflow in cfilelength! A file is too large (but I don't know which...).");
```
Comment (MageKing17) [inline code/cfile/cfile.cpp]: Checking maximum `size_t` to look for an `int` overflow again.

**Comment by asarium** [comment]
I have fixed the wrong Assertion

File: code/weapon/beam.cpp [hunk 4]
```diff
@@ -427,7 +427,7 @@ int beam_fire(beam_fire_info *fire_info)
 	}	
 
 	// create the associated object
-	objnum = obj_create(OBJ_BEAM, ((fire_info->shooter != NULL) ? OBJ_INDEX(fire_info->shooter) : -1), new_item - Beams, &vmd_identity_matrix, &vmd_zero_vector, 1.0f, OF_COLLIDES);
+	objnum = obj_create(OBJ_BEAM, ((fire_info->shooter != NULL) ? OBJ_INDEX(fire_info->shooter) : -1), (int)(new_item - Beams), &vmd_identity_matrix, &vmd_zero_vector, 1.0f, OF_COLLIDES);
```
Comment (MageKing17) [inline code/weapon/beam.cpp]: Would it be better to create a new `BEAM_INDEX` macro to be used here instead of `(int)(new_item - Beams)`?

**Comment by asarium** [comment]
Comments have been addressed and I also fixed a warning that happened if the code was compiled in 32-bit mode.

File: code/weapon/beam.cpp [hunk 5]
```diff
@@ -553,7 +553,7 @@ int beam_fire_targeting(fighter_beam_fire_info *fire_info)
 	// type c is a very special weapon type - binfo has no meaning
 
 	// create the associated object
-	objnum = obj_create(OBJ_BEAM, OBJ_INDEX(fire_info->shooter), new_item - Beams, &vmd_identity_matrix, &vmd_zero_vector, 1.0f, OF_COLLIDES);
+	objnum = obj_create(OBJ_BEAM, OBJ_INDEX(fire_info->shooter), (int)(new_item - Beams), &vmd_identity_matrix, &vmd_zero_vector, 1.0f, OF_COLLIDES);
```
Comment (MageKing17) [inline code/weapon/beam.cpp]: This line should also use the new `BEAM_INDEX` macro.

**Comment by MageKing17** [comment]
One last thing and then everything looks fine to me (although it's a very big diff and I could very well have missed something).

**Comment by asarium** [comment]
Thanks for reviewing the changes. I'll make sure that the next batch is smaller.

**Pull request merged**